src/views/Post/CreatePost.vue

```diff
@@ -44,33 +50,60 @@
         postData.value.shifts.splice(index, 1);
     };
 
-    function formatCurrency(event:any) {
-        const cleanedValue = event.target.value.replace(/\D/g, '');
-        const parsedValue = parseFloat(cleanedValue);
-        const formattedValue = parsedValue.toLocaleString('vi-VN', {
-            style: 'currency',
-            currency: 'VND',
-        });
-        postData.value.teachingFee = formattedValue;
+    function validateAgeRange() {
+        if (postData.value.requestTeacherAgeFrom && postData.value.requestTeacherAgeTo ) {
+            if (postData.value.requestTeacherAgeFrom >= postData.value.requestTeacherAgeTo) {
+                validationMessage.value = 'Tuổi bên trái phải nhỏ hơn Tuổi bên phải';
+                return false
+            } else {
+                validationMessage.value = '';
+                return true
+            }
+        } 
+        if (postData.value.requestTeacherAgeFrom || postData.value.requestTeacherAgeTo ) {
+            validationMessage.value = 'Tuổi bên trái và tuổi bên phải phải được chọn cả 2';
+                return false
+        }
+        return true
+    }
+
+    function isShiftValid(shift:ShiftInterface) {
+        return shift.weekday && shift.startTime && shift.endTime ? true : false
     }
 
-    function checkAgeRange() {
-        if (postData.value.requestFromAge && postData.value.requestToAge && postData.value.requestFromAge >= postData.value.requestToAge) {
-            validationMessage.value = 'Invalid age range';
-        } else {
-            validationMessage.value = '';
-        }
+    function isFormValid() {
+        console.log(isSubjectSelected.value, validateAgeRange(), postData.value.shifts.every((shift) => isShiftValid(shift)))
+        return isSubjectSelected.value 
+        && validateAgeRange()
+        && postData.value.shifts.every((shift) => isShiftValid(shift))
+        && postData.value.learnerName 
+        && postData.value.learnerSex
+        && postData.value.learnerbirthYear
+        && postData.value.teachingLocation
+        && postData.value.totalTeachingTime
+        && postData.value.totalTeachingTimeUnit
+        && postData.value.teachingFee
+        && postData.value.teachingFeeUnit
+        && postData.value.requestTeacherSex 
     }
 
-    function handleSubmit() {
-        // Access the form values
-        console.log(postData.value);
+    async function handleSubmit() {
+        const subjectIds: string[] = Object.keys(postData.value.subjects)
+        .filter((key) => postData.value.subjects[key].checked === true);
+        isSubjectSelected.value = subjectIds.length > 0;
+        isFormSubmitted.value = true;
+        if (isFormValid()) {
+            postData.value.subjects = subjectIds
+            const response = await PostService.createPost(postData.value);
+            if (response.status < 300) {
+                router.push('/');
+            }
+        }
```

Comment:
change async function handleSubmit() to
```suggestion
     async function handleSubmit() {
        isFormSubmitted.value = true;
        if (!isFormValid.value) {
            return
        }
        postData.value.subjects = subjectSelected.value;
        const response = await PostService.createPost(postData.value);
        if (response.status < 300) {
            router.push('/');
        }
    }
```